File: C/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string s = get_string("Input: ");
    int a = strlen(s);
    int b;
    int c = 0;
    int d;
    int e = 0;
    int g;
    double h = 0;

    for (int i = 0; i < a; i++)
    {
        if ((s[i] == '.') || (s[i] == ',') || (s[i] == ';') || (s[i] == '!') || (s[i] == '?') || ((char)s[i] == 39))
        {
        c++;
        }
    }

    for (d = 0; d < a; d++)
    {
        if (s[d] == ' ')
        {
            e++;
        }
    }

    for (g = 0; g < a; g++)
    {
        if ((s[g] == '.') || (s[g] == '!') || (s[g] == '?'))
        {
            h++;
        }
    }

    int w = (e+1);
    float f = a - (e + c);
    
    f /= w;
    f = f*100;
    h /= w;
    h = h*100;
    float index = (0.0588 * f - 0.296 * h - 15.8);
    index = round(index);
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index <= 15)
    {
        printf("Grade %.0f\n", index);
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
}

File: Python/readability.py
from cs50 import get_string

get = get_string("Input: \n")
periods = 0
sentences = 0
words = 0

for i in range(len(get)):
    if (get[i] == '.') or (get[i] == ',') or (get[i] == '?') or (get[i] == ';') or (get[i] == '!') or (get[i] == "'"):
        periods += 1
    if get[i] == ' ':
        words += 1
    if (get[i] == '.') or (get[i] == '?') or (get[i] == '!'):
        sentences += 1

f = (len(get) - (words + periods))
words += 1
f /= words
f *= 100
sentences /= words
sentences *= 100
index = (0.0588 * f - 0.296 * sentences - 15.8)

if index < 1:
    print("Before Grade 1")
elif index <= 15:
    print(f"Grade {(round(index))}")
elif index > 16:
    print("Grade 16+")